src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -1245,6 +1355,43 @@ DatabaseShardImp::finalizeShard(
                 return;
             it->second.state = ShardInfo::State::final;
             updateStatus(lock);
+
+            auto const boundaryIndex = getShardBoundaryIndex(lock);
+            auto const isHistoricalShard = shardIndex < boundaryIndex;
+
+            if (isHistoricalShard)
+            {
+                if (!historicalPaths_.empty() &&
+                    shard->getDir().parent_path() == dir_)
+                {
+                    // This is a historical shard that wasn't
+                    // placed at a separate historical path
+                    JLOG(j_.warn()) << "shard " << shardIndex
+                                    << " is not stored at a historical path";
+                }
+            }
+
+            else
+            {
+                // Not a historical shard. Shift recent shards
+                // if necessary
+                checkRecentShards(lock);
+                assert(shardIndex - boundaryIndex <= 1 || !boundaryIndex);
```

Comment:
I know it's just an assert, but when using short-circuit operators, I usually like to do the "easier" check first. So `assert(!boundaryIndex || shardIndex - boundaryIndex <= 1);`

Reply:
Done